strax/processing/peak_building.py

```diff
@@ -170,6 +180,14 @@ def store_downsampled_waveform(
             wv_buffer[: p["length"] * downsample_factor].reshape(-1, downsample_factor).sum(axis=1)
         )
         p["dt"] *= downsample_factor
+
+        # If the waveform is downsampled, we can store the first samples of the waveform
+        if store_waveform_start & (downsample_factor <= max_downsample_factor_waveform_start):
+            if p["length"] > len(p["data_start"]):
+                p["data_start"] = wv_buffer[: len(p["data_start"])]
+            else:
+                p["data_start"][: p["length"]] = wv_buffer[: p["length"]]
+
     else:
```

Comment:
Maybe here you should also assign `data_start`.

Reply:
Hi @dachengx, I think this case is not needed as we do not downsample the waveform. The `data` field contains all needed information.